python/emr/app.py

```diff
@@ -82,7 +84,7 @@ def __init__(
                 ),
             ),
             # note job_flow_role is an instance profile (not an iam role)
-            job_flow_role=emr_job_flow_profile.instance_profile_name,
+            job_flow_role=emr_job_flow_profile.instance_profile_name or "",
```

Comment:
Under what circumstances is this potentially `None`?

Reply:
Here is the specific mypy error we were seeing:
```
app.py:87: error: Argument "job_flow_role" to "CfnCluster" has incompatible type "Optional[str]"; expected "str"
```

Would adding a line like the following somewhere above this statement be acceptable?

```assert emr_job_flow_profile.instance_profile_name is not None```